common/txmgr/types/tx.go

```diff
@@ -26,3 +26,8 @@ const (
 	TxAttemptInsufficientEth = TxAttemptState("insufficient_eth")
 	TxAttemptBroadcast       = TxAttemptState("broadcast")
 )
+
+// Transaction is the type that callers get back, when they create a Transaction using the Txm.
+type Transaction interface {
+	GetID() string
```

Comment:
Drop the `Get` prefix?
```suggestion
	ID() string
```